### ISSUE TITLE
Hiddenservice v3

### ISSUE THREAD

**Issue by Hagge15** [torpyorg/torpy#15]
Since v2 is deprecated i can't use this anymore to connect to my hidden service. 
I really would like someone to create a torclient for v3. It is a really good package when it works and i'm just waiting for someone to create the v3 now.
But in the meantime, are their even any alternatives that exists?

**Comment by jbrown299** [comment]
Due to v2 deprecation timeline (https://blog.torproject.org/v2-deprecation-timeline) v2 will be deprecated from `July 15th, 2021`. So for now, everything is still working. But yes, we need to be prepared for changes.

**Comment by Artucuno** [comment]
So will v3 support be added soon?

**Comment by iven86** [comment]
Same here waiting for updates to support v3, any new update? Or if there another new version we can use it?

**Comment by jbrown299** [comment]
Working on it

**Comment by JamesHabben** [comment]
I don't see a new branch here in the repo. Is there anything that others can help on bringing v3 into this project?

**Comment by TJFJefroy** [comment]
looks like torpy is dead

**Comment by albertogeniola** [comment]
Any news on this? @jbrown299 : can we help somehow?

**Comment by Artucuno** [comment]
Welcome back @jbrown299 !

**Comment by Hammer2900** [comment]
Too bad the current version of this library does not work as needed now, probably because of the protocol version 3

**Comment by jbrown299** [comment]
Yes I have some progress on v3. Can commit if you wish to finish...15.12.2021, 22:57, "James Habben" ***@***.***>:
I don't see a new branch here in the repo. Is there anything that others can help on bringing v3 into this project?

**Comment by Artucuno** [comment]
@jbrown299 Would the branch be functional? I can possibly help with some things.

**Comment by FarmEquipment69** [comment]
> Yes I have some progress on v3. Can commit if you wish to finish...15.12.2021, 22:57, "James Habben" ***@***.***>: I don't see a new branch here in the repo. Is there anything that others can help on bringing v3 into this project? —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

Can you please commit your progress on v3

**Comment by Suryah07** [comment]
Updates???

**Comment by Arizel79** [comment]
Updates?

**Comment by SafwanLjd** [comment]
Old thread, but for anyone still searching: torquests handles this now. Pure Python, no daemon, and v3 onion services work. The API tracks requests closely, so there isn't much new to learn. https://github.com/SafwanLjd/torquests
